haystack/components/retrievers/sentence_window_retrieval.py

```diff
@@ -14,9 +14,41 @@ class SentenceWindowRetrieval:
     """
     A component that retrieves surrounding documents of a given document from the document store.
 
+    It relies on the `source_id` and on the `doc.meta['split_id']` to get the surrounding documents from the document.
     This component is designed to work together with one of the existing retrievers, e.g. BM25Retriever,
     EmbeddingRetriever. One of these retrievers can be used to retrieve documents based on a query and then use this
     component to get the surrounding documents of the retrieved documents.
+
+
+    Usage example:
+    ```python
+    from haystack import Document, Pipeline
+    from haystack.components.retrievers.in_memory import InMemoryBM25Retriever
+    from haystack.components.retrievers import SentenceWindowRetrieval
+    from haystack.components.preprocessors import DocumentSplitter
+    from haystack.document_stores.in_memory import InMemoryDocumentStore
+
+    splitter = DocumentSplitter(split_length=10, split_overlap=5, split_by="word")
+    text = (
+            "This is a text with some words. There is a second sentence. And there is also a third sentence. "
+            "It also contains a fourth sentence. And a fifth sentence. And a sixth sentence. And a seventh sentence"
+    )
+    doc = Document(content=text)
+    docs = splitter.run([doc])
+    doc_store = InMemoryDocumentStore()
+    doc_store.write_documents(docs["documents"])
+
+
+    rag = Pipeline()
+    rag.add_component("bm25_retriever", InMemoryBM25Retriever(doc_store, top_k=1))
+    rag.add_component("sentence_window_retriever", SentenceWindowRetrieval(document_store=doc_store, window_size=3))
```

Comment:
I would modify the window size in order to show an example where the final result does not correspond to the original doc. This would make the example more meaningful.

Reply:
agree 👍🏽